recipes/lora_dpo_distributed.py

```diff
@@ -526,11 +531,21 @@ def save_checkpoint(
                     }
                 )
 
-            self._checkpointer.save_checkpoint(
-                checkpoint_dict,
-                epoch=epoch,
-                intermediate_checkpoint=intermediate_checkpoint,
-            )
+            # If the option was True, save only the adapter except for the last epoch
+            is_intermediate_epoch = epoch + 1 < self.total_epochs
```

Comment:
Thanks so much for your contribution : )
Small q: do we want to support the case where merged weights are never saved, even in the final epoch, and let the user merge them? Might be helpful when working with particularly large models. @joecummings

Reply:
Yep, I think @SalmanMohammadi's point is inline with what I was thinking here. This does mean that we need to do a better job explaining how to merge weights or what exactly to do with them afterwards.

Reply:
Concretely, should we fix `adapter_only=self._save_adapter_only` here, and add something to the effect of:
```
if not is_intermediate_epoch:
    log.info("Saving final model checkpoint. Please note that you have set save_adapter_only=True, so only adapter weights will be saved. You will need to merge the adapter weights into your base model for further use. See {where do we point them for now??}")
```